protocol/messenger_mailserver_cycle.go

```diff
@@ -705,3 +688,23 @@ func (m *Messenger) SubscribeMailserverAvailable() chan struct{} {
 	m.mailserverCycle.availabilitySubscriptions = append(m.mailserverCycle.availabilitySubscriptions, c)
 	return c
 }
+
+func (m *Messenger) disconnectStorenodeIfRequired() error {
+	m.logger.Debug("wakuV2 storenode status verification")
+
+	if m.mailserverCycle.activeMailserver != nil {
+		// Check whether we want to disconnect the active storenode
+		if m.mailserverCycle.activeMailserver.FailedRequests >= mailserverMaxFailedRequests {
+			m.penalizeMailserver(m.mailserverCycle.activeMailserver.ID)
+			signal.SendMailserverNotWorking()
+			m.logger.Info("connecting too many failed requests")
+			m.mailserverCycle.activeMailserver.FailedRequests = 0
+			return m.connectToNewMailserverAndWait()
+		}
+	} else {
+		// No active storenode, find a new one
+		m.cycleMailservers()
+	}
```

Comment:
Can we please refactor this in a early return manner to make it more readable?

```suggestion
	if m.mailserverCycle.activeMailserver = nil {
		// No active storenode, find a new one
		m.cycleMailservers()
		return
	}
	
    // Check whether we want to disconnect the active storenode
    if m.mailserverCycle.activeMailserver.FailedRequests >= mailserverMaxFailedRequests {
	    m.penalizeMailserver(m.mailserverCycle.activeMailserver.ID)
	    signal.SendMailserverNotWorking()
	    m.logger.Info("connecting too many failed requests")
	    m.mailserverCycle.activeMailserver.FailedRequests = 0
	    return m.connectToNewMailserverAndWait()
    }
```